### PR TITLE
practice1

### PR DESCRIPTION
**Pull request by ddryzhov**
Added categories for books: categorycontroller, changed records to class for get and set methods, added new bookdto without categories + new category dto, changed book mapper + created category mapper, added categpry model + changed book model, added method to bookrepo to find all books by category id and changed + added new services for the rest of the logic + added liquibase

### CODE REVIEW

File: src/main/java/mate/academy/bookstore/model/Book.java [hunk 1]
```diff
@@ -44,4 +51,13 @@ public class Book {
...
     private boolean isDeleted = false;
+
+    @ManyToMany(fetch = FetchType.LAZY)
+    @JoinTable(
+            name = "book_category",
+            joinColumns = @JoinColumn(name = "book_id"),
+            inverseJoinColumns = @JoinColumn(name = "category_id"))
+    @ToString.Exclude
+    @EqualsAndHashCode.Exclude
+    private Set<Category> categories = new HashSet<>();
```
Comment (MAMentorFX) [inline src/main/java/mate/academy/bookstore/model/Book.java]: @ToString(exclude =
@EqualsAndHashCode(exclude = 
on class level, so you don't repeat those annotations for each and every field

File: src/main/java/mate/academy/bookstore/model/Category.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    @NotBlank
+    @Column(unique = true)
+    private String name;
+
+    private String description;
+
+    @ManyToMany(mappedBy = "categories", fetch = FetchType.LAZY)
+    @ToString.Exclude
+    @EqualsAndHashCode.Exclude
```
Comment (MAMentorFX) [inline src/main/java/mate/academy/bookstore/model/Category.java]: the same

File: src/main/java/mate/academy/bookstore/service/BookServiceImpl.java [hunk 3]
```diff
@@ -66,4 +74,18 @@ public List<BookDto> search(BookSearchParameters params, Pageable pageable) {
                 .map(bookMapper::toDto)
                 .toList();
     }
+
+    @Override
+    public List<BookDtoWithoutCategoryIds> findAllByCategoryId(Long id) {
+        return bookRepository.findAllByCategoryId(id).stream()
+                .map(bookMapper::toDtoWithoutCategories)
+                .toList();
```
Comment (MAMentorFX) [inline src/main/java/mate/academy/bookstore/service/BookServiceImpl.java]: Mapstruct supports mapping collections from the box

File: src/main/java/mate/academy/bookstore/service/CategoryServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+public class CategoryServiceImpl implements CategoryService {
+    private final CategoryRepository categoryRepository;
+    private final CategoryMapper categoryMapper;
+
+    @Override
+    public List<CategoryDto> findAll() {
+        return categoryRepository.findAll()
+                .stream()
+                .map(categoryMapper::toDto)
+                .toList();
```
Comment (MAMentorFX) [inline src/main/java/mate/academy/bookstore/service/CategoryServiceImpl.java]: Mapstruct supports mapping collections from the box

**Review by MAMentorFX — CHANGES_REQUESTED** [review]

**Review by sarakhmen — APPROVED** [review]

**Pull request merged**